controllers/btpoperator_controller.go

```diff
@@ -2070,3 +2188,223 @@ func (r *BtpOperatorReconciler) reconcileResourcesWithoutChangingCrState(ctx con
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) getDefaultCredentialsSecret(ctx context.Context) (*corev1.Secret, error) {
+	var defaultCredentialsSecret *corev1.Secret
+	secrets := &corev1.SecretList{}
+	if err := r.List(ctx, secrets, client.MatchingLabels{managedByLabelKey: operatorName}); err != nil {
+		return nil, fmt.Errorf("unable to list managed secrets: %w", err)
+	}
+	if len(secrets.Items) == 0 {
+		return nil, nil
+	}
+	for i, s := range secrets.Items {
+		if s.Name == sapBtpServiceOperatorSecretName && s.Namespace == r.previousCredentialsNamespace {
+			defaultCredentialsSecret = &secrets.Items[i]
+			break
+		} else if s.Name == sapBtpServiceOperatorSecretName {
+			defaultCredentialsSecret = &secrets.Items[i]
+		}
+	}
+	return defaultCredentialsSecret, nil
+}
+
+func (r *BtpOperatorReconciler) annotateSecret(ctx context.Context, s *corev1.Secret, key, value string) error {
+	annotations := s.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	if annotations[key] == value {
+		return nil
+	}
+	annotations[key] = value
+	s.SetAnnotations(annotations)
+	return r.Update(ctx, s, client.FieldOwner(operatorName))
+}
+
+func (r *BtpOperatorReconciler) getSapBtpServiceOperatorConfigMap(ctx context.Context) (*corev1.ConfigMap, error) {
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(ctx, client.ObjectKey{Namespace: ChartNamespace, Name: sapBtpServiceOperatorConfigMapName}, cm); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return cm, nil
+}
+
+func (r *BtpOperatorReconciler) deleteResourcesIfBtpManagerSecretChanged(ctx context.Context) error {
+	clusterIdSecret, err := r.getSecretByNameAndNamespace(ctx, sapBtpServiceOperatorClusterIdSecretName, r.credentialsNamespaceFromSapBtpServiceOperatorSecret)
+	if err != nil {
+		return err
+	}
+	pod, err := r.getSapBtpServiceOperatorPod(ctx)
+	if err != nil {
+		return err
+	}
+	credentialsSecret, err := r.getSecretByNameAndNamespace(ctx, sapBtpServiceOperatorSecretName, r.credentialsNamespaceFromSapBtpServiceOperatorSecret)
+	if err != nil {
+		return err
+	}
+
+	isCredentialsNamespaceChanged := r.credentialsNamespaceFromSapBtpServiceOperatorSecret != "" &&
+		r.credentialsNamespaceFromSapBtpManagerSecret != r.credentialsNamespaceFromSapBtpServiceOperatorSecret
+
+	isClusterIdChanged := r.clusterIdFromSapBtpServiceOperatorConfigMap != "" &&
+		(r.clusterIdFromSapBtpManagerSecret != r.clusterIdFromSapBtpServiceOperatorConfigMap ||
+			r.clusterIdFromSapBtpServiceOperatorConfigMap != r.clusterIdFromSapBtpServiceOperatorClusterIdSecret)
+
+	if isCredentialsNamespaceChanged || isClusterIdChanged {
+		if clusterIdSecret != nil {
+			if err := r.deleteObject(ctx, clusterIdSecret); err != nil {
+				return err
+			}
+		}
+		if pod != nil {
+			if err := r.deleteObject(ctx, pod); err != nil {
+				return err
+			}
+		}
+	}
+
+	if isCredentialsNamespaceChanged {
```

Comment:
Could we flatten it to only two levels?
IMHO it would be more comprehensible.
```
if isCredentialsNamespaceChanged && credentialsSecret != nil {
```

Reply:
Of course, we don't need two ifs here.